### ISSUE TITLE
Posting status with only image attachment returns 422 sometimes

### ISSUE THREAD

**Issue by connyduck** [mastodon/mastodon#21864]
### Steps to reproduce the problem

1. Upload a image (I used a 6mb photo) with POST /api/v2/media. Get back status 200 and an id.
2. POST /api/v1/statuses with empty status and warningText and mediaIds=[receivedId]
3. in about 25% of cases, get back 422 instead of 200. 
...

### Expected behaviour

Either the media upload returns 202 so I know I have to wait until attaching the media id to a post, or POST /api/v1/statuses accepts the media id

### Actual behaviour

Attaching media to a status immediately after upload fails sometimes, although upload returned 200

### Specifications

Mastodon v4.0.2 on mastodon.social

**Comment by mcclure** [comment]
I want to note that the rare case connyduck mentions (the post successfully goes up, but with the attachment stripped away) is something I've managed to reproduce twice in live server tests, once on Oct 31 when mastodon.social was under heavy load and once yesterday when I was uploading a large image. In my opinion this is a **very** bad bug symptom, because the user's content goes up on the feed in a different form than they intended without the user being specifically notified (this could lead to misunderstandings etc), and there seems to be nothing we can do on the Tusky side to prevent the bad outcome because we are just following the API as documented.

**Comment by untitaker** [comment]
Is it possible that Tusky, as part of this procedure, retries the POST to `/api/v1/statuses`, and actually double-posts statuses?

I've seen this problem with Hachyderm where the database was so slow that the code implementing idempotency keys encountered race conditions, leading to the double-creation of statuses. See #21840 

In that scenario the first status will get all the media attached to it, and the second status will not get any media because all of it has already a non-null `status_id`.

**Comment by connyduck** [comment]
No, there is no retrying involved in this bug or I would have mentioned it. Status posting multiple times despite idempotency key set could have a similar cause though (database write still not recorded while another thing already reads)
